lib/containers/k8s.pm

```diff
@@ -21,19 +21,28 @@ use registration qw(add_suseconnect_product get_addon_fullname);
 our @EXPORT = qw(install_k3s uninstall_k3s install_kubectl install_helm install_oc);
 
 sub install_k3s {
-    my $k3s_version = get_var("CONTAINERS_K3S_VERSION", "v1.23.6+k3s1");
-    record_info('k3s', $k3s_version);
-    assert_script_run("curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION=$k3s_version sh -");
-    enter_cmd("k3s server &");
+  # Apply additional options. For more information see https://rancher.com/docs/k3s/latest/en/installation/install-options/#options-for-installation-with-script
+    if (get_var("CONTAINERS_K3S_VERSION")) {
+        my $k3s_version = get_var("CONTAINERS_K3S_VERSION");
+        record_info('k3s', $k3s_version);
+        assert_script_run("export INSTALL_K3S_VERSION=$k3s_version");
+    }
+    assert_script_run("export INSTALL_K3S_SYMLINK=" . get_var('K3S_SYMLINK')) if (get_var('K3S_SYMLINK'));
+    assert_script_run("export INSTALL_K3S_BIN_DIR=" . get_var('K3S_BIN_DIR')) if (get_var('K3S_BIN_DIR'));
+    assert_script_run("export INSTALL_K3S_CHANNEL=" . get_var('K3S_CHANNEL')) if (get_var('K3S_CHANNEL'));
+    assert_script_run("curl -sfL https://get.k3s.io | sh -");
+    # Note: The install script starts a k3s-server by default, unless INSTALL_K3S_SKIP_START is set to true
+    sleep(20);    # Wait one iteration interval before checking because the server needs some time to boot-up
     script_retry("test -e /etc/rancher/k3s/k3s.yaml", delay => 20, retry => 10);
+    assert_script_run('systemctl is-active k3s');
     assert_script_run("k3s kubectl get node");
-    script_run("mkdir \$HOME/.kube");
-    script_run("rm \$HOME/.kube/config");
+    assert_script_run("mkdir -p ~/.kube");
```

Comment:
I  think this one and the following cannot be asset_script_run... we don't know if the directory exists previously or not. The same for the config file. Please run the helm tests to check it.

Reply:
`mkdir -p` also returns 0 if the directory already exists, and `rm -f` also returns 0 if the file does not exists. But if you don't like it we can also use `script_run` here.